ignite/engine/engine.py

```diff
@@ -833,12 +834,17 @@ def _run_once_on_dataset(self) -> float:
                     # Should exit while loop if we can not iterate
                     if should_exit:
                         if not self._is_done(self.state):
+                            total_iters = (
+                                self.state.epoch_length * self.state.max_epochs
+                                if self.state.max_epochs is not None
+                                else None
```

Comment:
Normally, either `state.max_epochs` or `state.max_iters` is defined. We can set `else self.state.max_iters` instead of None.